src/webgpu/api/validation/image_copy/texture_related.spec.ts

```diff
@@ -175,6 +175,73 @@ g.test('mip_level')
     );
   });
 
+g.test('format')
+  .desc(`Test that it must be a full copy if the texture's format is depth/stencil format`)
+  .params(u =>
+    u //
+      .combine('method', kImageCopyTypes)
+      .combineWithParams([
+        { depthOrArrayLayers: 1, dimension: '2d' },
+        { depthOrArrayLayers: 3, dimension: '2d' },
+        { depthOrArrayLayers: 32, dimension: '3d' },
+      ] as const)
+      .combine('format', kSizedTextureFormats)
+      .filter(({ dimension, format }) => textureDimensionAndFormatCompatible(dimension, format))
+      .filter(formatCopyableWithMethod)
+      .beginSubcases()
+      .combine('mipLevel', [0, 2])
+      .combine('copyWidthModifier', [0, -1])
+      .combine('copyHeightModifier', [0, -1])
+      .expand('copyDepthModifier', ({ dimension: d }) => (d === '3d' ? [0, -1] : [0]))
+  )
+  .fn(async t => {
+    const {
+      method,
+      depthOrArrayLayers,
+      dimension,
+      format,
+      mipLevel,
+      copyWidthModifier,
+      copyHeightModifier,
+      copyDepthModifier,
+    } = t.params;
+
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    const size = { width: 32, height: 32, depthOrArrayLayers };
+    const texture = t.device.createTexture({
+      size,
+      dimension,
+      format,
+      mipLevelCount: 5,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+    });
+
+    let success = true;
+    if (
+      (info.depth || info.stencil) &&
+      (copyWidthModifier !== 0 || copyHeightModifier !== 0 || copyDepthModifier !== 0)
+    ) {
+      success = false;
+    }
+
+    const copySize = [
+      (size.width >> mipLevel) + copyWidthModifier * info.blockWidth,
+      (size.height >> mipLevel) + copyHeightModifier * info.blockHeight,
+      // Note that compressed format is not supported for 3D textures yet, so there is no info.blockDepth.
+      dimension === '3d'
+        ? (size.depthOrArrayLayers >> mipLevel) + copyDepthModifier
+        : size.depthOrArrayLayers,
```

Comment:
Shouldn't this still have the copyDepthModifier applied?

Reply:
If it is 2D array texture, copyDepthModifier will always be 0 (I added some comments for it at copyDepthModifier). Adding it or not is the same.

Reply:
Ah I see, sorry I didn't understand this.